jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/ZipUtil.java

```diff
@@ -63,6 +63,7 @@ public static void unzip(Path archive, Path destination) throws IOException {
             ByteStreams.copy(zipIn, out);
           }
         }
+        Files.setLastModifiedTime(entryPath, entry.getLastModifiedTime());
```

Comment:
Maybe we can even set the time when calling `createDirectories()` or `newOutputStream()` through `FileAttribute`? If that's possible, we can reduce two IO operations into one. Actually, it is necessary if `createDirectories()` creates multiple directories?

Reply:
I agree with you. Unfortunately, setting the `"basic:lastModifiedTime"` attribute when calling `createDirectories()` results in an [`UnsupportedOperationException`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html) which occurs when an attribute can't be set atomically while creating a directory so I'm not sure if it is possible.

Reply:
Hmm... maybe we need to to roll our own version of `Files.createDirectories()` that incrementally creates subdirectories when they don't exists and sets the file timestamp. Otherwise, we can still end up creating directories with the current timestamp. Thoughts?